### PR TITLE
WIP:Create app with protected user data

### PR DESCRIPTION
**Pull request by Rick-Anderson**

### CODE REVIEW

File: aspnet/security/authorization/secure-data.rst [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+.. literalinclude:: secure-data/samples/final/Controllers/ContactsController.cs
+  :language: c#
+  :start-after: snippet_ContactsController
+  :end-before: #endregion
+  :dedent: 4
+  :emphasize-lines: 4,5,9,10,13,14
+
+Add the ``ContactOperationsRequirements`` class to the *Authorization* folder to contain the requirements our app supports:
+
+.. literalinclude:: secure-data/samples/final/Authorization/ContactOperationsRequirements.cs
```
Comment (Rick-Anderson) [inline aspnet/security/authorization/secure-data.rst]: Per blowdart: (from PR #1734)
This appears to be mixed up operations and permissions. For example you've defined the name of the create operation as canCreate, which is the actual permission. This isn't what I'd expect, I'd expect the operation name to be create, and then you have a mapping table to map the permission role or claim to the actual operation name inside the handler, or in a helper.

**Comment by Rick-Anderson** [comment]
@joeaudette any chance you could help me finish this off. I can give you the details on a minor security code fix [@]blowdart asked me to apply

**Comment by joeaudette** [comment]
@Rick-Anderson glad to help in any way I can, not clear to me what you would like me to do?

**Comment by Rick-Anderson** [comment]
@joeaudette  I'll get the detailed request and let you know, probably in a couple weeks.

**Comment by Rick-Anderson** [comment]
This is what [@]blowdart reported:

```
using Microsoft.AspNetCore.Authorization.Infrastructure;
 
// Add https://github.com/blowdart/AspNetAuthorization-Samples/blob/master/src/AspNetAuthorization/Authorization/DocumentAuthorizationHandler.cs
// MapClaimsToOperations
 
namespace ContactManager.Authorization
{
    public static class ContactOperationsRequirements
    {
        public static OperationAuthorizationRequirement Create =  // Add a create handler
            new OperationAuthorizationRequirement { Name = "Create" };
        public static OperationAuthorizationRequirement Read =
            new OperationAuthorizationRequirement { Name = "Read" };  // Add a read handler
        public static OperationAuthorizationRequirement Update =
            new OperationAuthorizationRequirement { Name = "Update" }; // Add a delete handler
        public static OperationAuthorizationRequirement Delete =
            new OperationAuthorizationRequirement { Name = Constants.canDelete };
        public static OperationAuthorizationRequirement ContainsOne =
          new OperationAuthorizationRequirement { Name = Constants.ContainsOne };
    }
 
    public class Constants
    {
        public static readonly string canDelete = "canDelete";
        public static readonly string ContainsOne = "ContainsOne";
    }
}
```
// operations are not permission, permissions are what grant operations
This appears to be mixed up operations and permissions. For example you've defined the name of the create operation as canCreate, which is the actual permission. This isn't what I'd expect, I'd expect the operation name to be create, and then you have a mapping table to map the permission role or claim to the actual operation name inside the handler, or in a helper.
 
 
He said I need to do what he did in https://github.com/blowdart/AspNetAuthorization-Samples/blob/master/src/AspNetAuthorization/Authorization/DocumentAuthorizationHandler.cs

```
private IEnumerable<OperationAuthorizationRequirement> MapClaimsToOperations(string claimValue)
        {
            var mappedPermissions = new List<OperationAuthorizationRequirement>();
 
            if (claimValue.Contains('C'))
            {
                mappedPermissions.Add(Operations.Create);
            }
            if (claimValue.Contains('R'))
            {
                mappedPermissions.Add(Operations.Read);
            }
            if (claimValue.Contains('U'))
            {
                mappedPermissions.Add(Operations.Update);
            }
            if (claimValue.Contains('D'))
            {
                mappedPermissions.Add(Operations.Delete);
            }
 
            return mappedPermissions;
        }
```
Not sure how much value adding create/read/delete handlers would add, but Barry said think about that.

**Comment by Rick-Anderson** [comment]
@joeaudette do you understand the code changes requested?

**Comment by Rick-Anderson** [comment]
@joeaudette let me get the code converted to VS 2017 and send you a word doc of the tutorial.

**Comment by Rick-Anderson** [comment]
moved to https://github.com/aspnet/Docs/pull/3225

**Pull request closed without merging**